### PR TITLE
Fixing issue #1245 as well as reverting a recent change to the roll mechanic

### PR DESCRIPTION
**Pull request by matt-bel**
Feel free to ignore the first part of the pull request, but I believe I have found a passable work-around for the tank firing bug.

### CODE REVIEW

**Comment by Liruxo** [comment]
First up: Thank you for fixing these issues and contributing to FlansMod. However, i have a few problems with this PR:
- The Rapid-Fire fix seems good, so please create a new PR only containing the "Fixing Rapid-fire bug" (a2594ed) commit. I do not see any problems with it, so it should get merged relatively soon. (Unless @ChrisLane has some criticism)

The other commit is in my opinion problematic for a few reasons:
- Referencing the comment "Roll in third person has been the default in FlansMod and the recent change has been extremely disorienting.": This comment is in it current form completely irrelevant, as it references the previous comment. which is not helpful or explains the code in the line below. I would like to see something like "Roll is disorientating when the player is not controlling the vehicle" as it accurately represents to code and helps people not familiar understand its purpose.
-The commit message " Revert the unnecessary changes to camera rotation in planes. This isn't warthunder or a plane sim, people are" is also very opinionated and contains unnecessary information. The message "Revert the changes to camera rotation in planes" is completely fine and descriptive enough, please try to use similar messages in the future

To sum up: Please do not use opinionated and undescriptive text in source code comments and commit messages. You are free (and encouraged) to explain your reasoning in the PR comments, as this is the right place to do so

Continuing, i have additional problems with the current solution you proposed: Ideally i would like a config based solution, containing a few more options allowing players to choose the right setting for them. For example i personally would also like to experience roll as passenger. However, i am open for other solutions so please open another PR for this issue so we can discuss the issue there.

**Comment by ChrisLane** [comment]
I think perhaps you haven't fully understood what your changes do. There are still instances where the same behaviour will occur.
The existing code was already a bit of a mess to follow and so I've done some refactoring and fixed #1245 as well. Perhaps you could test the changes and see if you're happy with them.

Sorry it's taken me so long to respond to this. As @Liruxo mentioned, PRs should ideally focus on one thing to make it easier to review and discuss and so I will be closing this.
